scene/resources/world_boundary_shape_2d.cpp

```diff
@@ -86,10 +87,27 @@ void WorldBoundaryShape2D::draw(const RID &p_to_rid, const Color &p_color) {
 	Vector2 point = distance * normal;
 	real_t line_width = 3.0;
 
-	Vector2 l1[2] = { point - normal.orthogonal() * 100, point + normal.orthogonal() * 100 };
-	RS::get_singleton()->canvas_item_add_line(p_to_rid, l1[0], l1[1], p_color, line_width);
-	Vector2 l2[2] = { point + normal.normalized() * (0.5 * line_width), point + normal * 30 };
-	RS::get_singleton()->canvas_item_add_line(p_to_rid, l2[0], l2[1], p_color, line_width);
+	// Draw collision shape line.
+	PackedVector2Array points = {
+		point - normal.orthogonal() * 100,
+		point - normal.orthogonal() * 60,
+		point + normal.orthogonal() * 60,
+		point + normal.orthogonal() * 100
+	};
+
+	Color transparent_colour = Color(p_color.r, p_color.g, p_color.b, 0);
+	PackedColorArray colours = {
+		transparent_colour,
+		p_color,
+		p_color,
+		transparent_colour
+	};
+
+	RS::get_singleton()->canvas_item_add_polyline(p_to_rid, points, colours, line_width);
+
+	// Draw normal line.
+	Vector2 l2[2] = { Vector2(0, 0), point + normal * 30 };
+	RS::get_singleton()->canvas_item_add_line(p_to_rid, l2[0], l2[1], Color::named("BLACK"), 1);
```

Comment:
```suggestion
	RS::get_singleton()->canvas_item_add_line(p_to_rid, l2[0], l2[1], Color(), 1);
```
Equivalent, and using named colour access is not recommended for performance unless necessary